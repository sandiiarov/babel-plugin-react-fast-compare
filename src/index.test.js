const { transform } = require('@babel/core');

const config = {
  plugins: [require('./index')],
  babelrc: false,
  configFile: false,
};

test('with default import', () => {
  const source = `
    import _ from 'lodash';

    _.capitalize('foo');
    _.isEqual({ foo: 'baz', baz: 'foo' });
  `;

  expect(transform(source, config).code).toMatchSnapshot();
});

test('with named import', () => {
  const source = `
    import { capitalize, isEqual } from 'lodash';

    capitalize('foo');
    isEqual({ foo: 'baz', baz: 'foo' });
  `;

  expect(transform(source, config).code).toMatchSnapshot();
});

test('with default and named import', () => {
  const source = `
    import _, { isEqual } from 'lodash';

    _.capitalize('foo');
    isEqual({ foo: 'baz', baz: 'foo' });
  `;

  expect(transform(source, config).code).toMatchSnapshot();
});

test('with react-fast-compare', () => {
  const source = `
    import { capitalize } from 'lodash';
    import isEqual from 'react-fast-compare';

    _.capitalize('foo');
    isEqual({ foo: 'baz', baz: 'foo' });
  `;

  expect(transform(source, config).code).toMatchSnapshot();
});

test('with default import from lodash/isEqual', () => {
  const source = `
    import _capitalize from 'lodash/capitalize';
    import _isEqual from 'lodash/isEqual';

    _capitalize('foo');
    _isEqual({ foo: 'baz', baz: 'foo' });
  `;

  expect(transform(source, config).code).toMatchSnapshot();
});

test('with default import from lodash/lang/isEqual', () => {
  const source = `
    import _capitalize from 'lodash/capitalize';
    import _isEqual from 'lodash/isEqual';

    _capitalize('foo');
    _isEqual({ foo: 'baz', baz: 'foo' });
  `;

  expect(transform(source, config).code).toMatchSnapshot();
});
