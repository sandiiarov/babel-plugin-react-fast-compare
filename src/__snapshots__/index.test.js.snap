// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`with default and named import 1`] = `
"import isEqual from \\"react-fast-compare\\";
import _ from 'lodash';

_.capitalize('foo');

isEqual({
  foo: 'baz',
  baz: 'foo'
});"
`;

exports[`with default import 1`] = `
"import isEqual from \\"react-fast-compare\\";
import _ from 'lodash';

_.capitalize('foo');

isEqual({
  foo: 'baz',
  baz: 'foo'
});"
`;

exports[`with default import from lodash/isEqual 1`] = `
"import isEqual from \\"react-fast-compare\\";
import _capitalize from 'lodash/capitalize';

_capitalize('foo');

isEqual({
  foo: 'baz',
  baz: 'foo'
});"
`;

exports[`with default import from lodash/lang/isEqual 1`] = `
"import isEqual from \\"react-fast-compare\\";
import _capitalize from 'lodash/capitalize';

_capitalize('foo');

isEqual({
  foo: 'baz',
  baz: 'foo'
});"
`;

exports[`with named import 1`] = `
"import isEqual from \\"react-fast-compare\\";
import { capitalize } from 'lodash';
capitalize('foo');
isEqual({
  foo: 'baz',
  baz: 'foo'
});"
`;

exports[`with react-fast-compare 1`] = `
"import { capitalize } from 'lodash';
import isEqual from 'react-fast-compare';

_.capitalize('foo');

isEqual({
  foo: 'baz',
  baz: 'foo'
});"
`;
